custom_components/tuya_local/devices/royalclima_fresh_climate.yaml

```diff
@@ -0,0 +1,409 @@
+        - dps_val: mid_low
+          value: medlow
+        - dps_val: mid
+          value: medium
+        - dps_val: mid_high
+          value: medhigh
+        - dps_val: high
+          value: high
+        - dps_val: strong
+          value: strong
+    - id: 102
+      name: preset_mode
+      type: string
+      translation_key: Recirculate
+      mapping:
+        - dps_val: "off"
+          value: "Off"
+        - dps_val: low
+          value: Fresh Air Low
+        - dps_val: mid
+          value: Fresh Air Medium
+        - dps_val: strong
+          value: Fresh Air High
+        - dps_val: auto
+          value: Fresh Air Auto
+    - id: 20
+      type: bitfield
+      name: fault_code
+    - id: 105
+      type: string
+      name: sleep_mode
+    - id: 110
+      type: bitfield
+      name: flags
+    - id: 113
+      type: string
+      name: swing_mode
+      mapping:
+        - dps_val: "0"
+          constraint: horizontal_swing
+          conditions:
+            - dps_val: "0"
+              value: "off"
+            - dps_val: "1"
+              value: horizontal
+            - value: horizontal
+        - dps_val: "1"
+          constraint: horizontal_swing
+          conditions:
+            - dps_val: "0"
+              value: vertical
+            - dps_val: "1"
+              value: both
+            - value: both
+        - constraint: horizontal_swing
+          conditions:
+            - dps_val: "0"
+              value: vertical
+            - value: both
+    - id: 114
+      type: string
+      name: horizontal_swing
+      hidden: true
+    - id: 119
+      type: string
+      name: electricity_management
+    - id: 120
+      type: string
+      name: gen_mode
+    - id: 123
+      type: hex
+      name: flags_2
+    - id: 125
+      type: string
+      name: air_quality
+      optional: true
+    - id: 128
+      type: string
+      name: model_code
+    - id: 129
+      type: string
+      name: energy
+    - id: 130
+      type: integer
+      name: eco_temp
+    - id: 132
+      type: boolean
+      name: hot_cool
+    - id: 133
+      type: string
+      name: swing_action
+    - id: 134
+      type: json
+      name: statistics
+secondary_entities:
+  - entity: fan
+    name: Fresh air
+    translation_key: ventilation
+    icon: "mdi:air-purifier"
+    dps:
+      - id: 102
+        name: switch
+        type: string
+        mapping:
+          - dps_val: "off"
+            value: false
+          - dps_val: "auto"
+            value: true
+  - entity: select
+    name: Fresh mode
+    dps:
+      - id: 102
+        name: preset_mode
+        type: string
+        mapping:
+          - dps_val: "off"
+            value: "Off"
+          - dps_val: low
+            value: Fresh Air Low
+          - dps_val: mid
+            value: Fresh Air Medium
+          - dps_val: strong
+            value: Fresh Air High
+          - dps_val: auto
+            value: Fresh Air Auto
+  - entity: sensor
```

Comment:
This is already available in the climate entity, so does not need to be duplicated as a sensor